CRM/Member/ActionMapping.php

```diff
@@ -91,12 +91,15 @@ public function createQuery($schedule, $phase, $defaultParams) {
     }
 
     // FIXME: Numbers should be constants.
+    $contribution_status_id_pending = CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'contribution_status_id', 'Pending');
     if (in_array(2, $selectedStatuses)) {
       //auto-renew memberships
-      $query->where("e.contribution_recur_id IS NOT NULL");
+      $query->join('contribution_recur', 'inner join civicrm_contribution_recur on e.contribution.id = contribution_recur.id');
+      $query->where("contribution_recur.contribution_status_id = #pending", ['pending' => $contribution_status_id_pending]);
     }
     elseif (in_array(1, $selectedStatuses)) {
-      $query->where("e.contribution_recur_id IS NULL");
+      $query->join('contribution_recur', 'inner join civicrm_contribution_recur on e.contribution.id = contribution_recur.id');
```

Comment:
@stesi561 I think this needs to be a LEFT JOIN as we want the membership reminders in this situation to go to any memberships that don't have a contribution_recur_id OR where there is one and it is is not in progress right?

Reply:
correct!

Reply:
I guess we also need to account for null values as well! :)